velox/functions/lib/LikeFunctionUtils.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <velox/type/StringView.h>
+#include <velox/vector/TypeAliases.h>
+#include <string>
+
+namespace facebook::velox::functions {
+namespace likeutils {
+
+// This function checks if the string 'pattern' has wildcard characters ('_' or
+// '%'). It returns true if wildcard characters are found, else false.
+bool isWildcardInPattern(const StringView& pattern);
+
+// This function sequentially matches string input to a fixed pattern (with no
+// wildcard characters).
+bool matchExactPattern(const StringView& input, const StringView& pattern);
+
+// This function checks if the string is a 'prefixPattern' (which is a fixed
+// pattern followed by one or more '%'), returns the prefix pattern's length.
```

Comment:
What does this function return if 'pattern' is '%'? Do we have an optimization for that pattern? It would be helpful to explain what is the return value in case the pattern is not a prefix pattern.

Reply:
Thanks for the suggestion, updated the comment. If pattern is %, it is categorized as a prefixPattern (could either be a prefix or suffix pattern but the order of checks makes it a prefix pattern). There is no specific optimization for this case.

Reply:
Thank you for clarifying. Would be nice to optimize this case as well. It would help with queries auto-generated by tools.

Reply:
Thanks for the suggestion, added a case for patterns with just wildcard characters.